core/contracts/primitive_stream.kf

```diff
@@ -312,22 +312,27 @@ procedure insert_record($date_value text, $value decimal(21,3)) public {
 // Truflations calculation is ((current_primitive/first_primitive)*100).
 // This essentially gives us the same result, but with an extra 3 digits of precision.
 // index := (currentPrimitive * 100) / basePrimitive
-procedure get_index($date_from text, $date_to text, $frozen_at int) public view returns table(
+procedure get_index($date_from text, $date_to text, $frozen_at int, $base_date text) public view returns table(
         date_value text,
         value decimal(21,3)
         ) {
+    if $base_date IS NULL OR $base_date == '' {
+        $base_date := '2010-01-01';
+    }
 
-    $baseValue decimal(21,3) := get_base_value();
+    $baseValue decimal(21,3) := get_base_value($base_date);
     if $baseValue == 0::decimal(21,3) {
         error('base value is 0');
     }
 
     return SELECT date_value, (value * 100::decimal(21,3)) / $baseValue as value FROM get_record($date_from, $date_to, $frozen_at);
 }
 
-// get_base_value returns the first value of the primitive stream
-procedure get_base_value() private view returns (value decimal(21,3)) {
-    for $row in SELECT value from primitive_events ORDER BY date_value ASC, created_at DESC LIMIT 1 {
+// get_base_value returns the first value of the primitive stream after the given date
+procedure get_base_value($base_date text) private view returns (value decimal(21,3)) {
+    for $row in SELECT value from primitive_events
+    WHERE date_value >= $base_date
```

Comment:
just a check: if `base_date` is null, this works normally then, correct?

Reply:
Yes, it will use 2010-01-01